runelite-client/pom.xml

```diff
@@ -448,17 +448,32 @@
 			<plugin>
 				<groupId>org.apache.maven.plugins</groupId>
 				<artifactId>maven-pmd-plugin</artifactId>
-				<version>3.16.0</version>
+				<version>3.21.2</version>
 				<dependencies>
+					<dependency>
+						<groupId>net.sourceforge.pmd</groupId>
+						<artifactId>pmd-compat6</artifactId>
+						<version>7.0.0</version>
+					</dependency>
 					<dependency>
 						<groupId>net.sourceforge.pmd</groupId>
 						<artifactId>pmd-core</artifactId>
-						<version>6.44.0</version>
+						<version>7.0.0</version>
 					</dependency>
 					<dependency>
 						<groupId>net.sourceforge.pmd</groupId>
 						<artifactId>pmd-java</artifactId>
-						<version>6.44.0</version>
+						<version>7.0.0</version>
+					</dependency>
+					<dependency>
+						<groupId>net.sourceforge.pmd</groupId>
+						<artifactId>pmd-javascript</artifactId>
```

Comment:
?

Reply:
I meant to comment or add more info in the commit message, but [maven-pmd-plugin depends on these](https://github.com/apache/maven-pmd-plugin/blob/a3ac53c954b00527d17603c8179d6555b14124d9/pom.xml#L174-L183), and I think with the API changes from 6 to 7 `pmd-compat6` needs all of these updated and available to work properly. Their docs for [upgrading PMD at runtime](https://maven.apache.org/plugins/maven-pmd-plugin/examples/upgrading-PMD-at-runtime.html) also show upgrading these packages too.

Reply:
It looks like maven-pmd-plugin has updated to pmd 7 now as of Apr 18, does this mean it is no longer necessary to include all of these irrelevant dependencies?

Reply:
> It looks like maven-pmd-plugin has updated to pmd 7 now as of Apr 18, does this mean it is no longer necessary to include all of these irrelevant dependencies?

It seems like it is no longer required.